katdal/h5datav2_5.py

```diff
@@ -0,0 +1,631 @@
+"""Data accessor class for HDF5 files produced by AVN DBE."""
+
+import logging
+
+import numpy as np
+import h5py
+import katpoint
+
+from .dataset import DataSet, WrongVersion, BrokenFile, Subarray, \
+    DEFAULT_SENSOR_PROPS, DEFAULT_VIRTUAL_SENSORS, _robust_target
+from .spectral_window import SpectralWindow
+from .sensordata import RecordSensorData, SensorCache
+from .categorical import CategoricalData, sensor_to_categorical
+from .lazy_indexer import LazyIndexer, LazyTransform
+
+logger = logging.getLogger(__name__)
+
+# Simplify the scan activities to derive the basic state of the antenna (slewing, scanning, tracking, stopped)
+SIMPLIFY_STATE = {'scan_ready': 'slew', 'scan': 'scan', 'scan_complete': 'scan', 'track': 'track', 'slew': 'slew'}
+
+SENSOR_PROPS = dict(DEFAULT_SENSOR_PROPS)
+SENSOR_PROPS.update({
+    '*activity': {'greedy_values': ('slew', 'stop'), 'initial_value': 'slew',
+                  'transform': lambda act: SIMPLIFY_STATE.get(act, 'stop')},
+    '*target': {'initial_value': '', 'transform': _robust_target},
+    # These float sensors are actually categorical by nature as they represent user settings
+    'RFE/center-frequency-hz': {'categorical': True},
+    'RFE/rfe7.lo1.frequency': {'categorical': True},
+    '*attenuation': {'categorical': True},
+    '*attenuator.horizontal': {'categorical': True},
+    '*attenuator.vertical': {'categorical': True},
+    'RFE/rfe.band.select.LCP': {'categorical': True},
```

Comment:
Are these really floats?

Reply:
This might be some left-overs from the old v2 file that I copied. I might need to check that I understand properly what's going on there.

Reply:
Corrected in c2daa49.